This patch should let you compile gettext on FreeBSD.


To apply this in the gettext sources:

# cd /usr/local/src/gettext-0.10.35
(or wherever you have extracted the source tree)
# patch -p1 </PATH/TO/gettext-0.10.35-3-memcpy.diff


diff -Nurp gettext-0.10.35.kn.2/intl/ChangeLog gettext-0.10.35.kn.3/intl/ChangeLog
--- gettext-0.10.35.kn.2/intl/ChangeLog	Wed Apr 29 16:43:14 1998
+++ gettext-0.10.35.kn.3/intl/ChangeLog	Wed Sep 23 18:18:45 1998
@@ -1,3 +1,8 @@
+1998-09-23  Kalle Olavi Niemitalo  <tosi@stekt.oulu.fi>
+
+	* localealias.c: Check #ifndef HAVE_MEMCPY rather than #ifndef
+	memcpy.  Patch from Paul Thomas.
+
 1998-04-29  Ulrich Drepper  <drepper@cygnus.com>
 
 	* intl/localealias.c (read_alias_file): Use unsigned char for
diff -Nurp gettext-0.10.35.kn.2/intl/localealias.c gettext-0.10.35.kn.3/intl/localealias.c
--- gettext-0.10.35.kn.2/intl/localealias.c	Wed Apr 29 16:42:42 1998
+++ gettext-0.10.35.kn.3/intl/localealias.c	Wed Sep 23 18:18:45 1998
@@ -59,7 +59,7 @@ void free ();
 # include <string.h>
 #else
 # include <strings.h>
-# ifndef memcpy
+# ifndef HAVE_MEMCPY
 #  define memcpy(Dst, Src, Num) bcopy (Src, Dst, Num)
 # endif
 #endif
diff -Nurp gettext-0.10.35.kn.2/m4/ChangeLog gettext-0.10.35.kn.3/m4/ChangeLog
--- gettext-0.10.35.kn.2/m4/ChangeLog	Wed Sep 23 17:05:26 1998
+++ gettext-0.10.35.kn.3/m4/ChangeLog	Wed Sep 23 18:20:10 1998
@@ -5,6 +5,9 @@
 	together with AC_CHECK_LIB because that has its own caching.  Set
 	INTLLIBS=-lintl if gettext() and bindtextdomain() are in -lintl.
 	Use $INTLLIBS when checking for dcgettext() and _nl_msg_cat_cntr.
+ 
+ 	* gettext.m4 (AM_GNU_GETTEXT): Check whether memcpy() exists.
+ 	Reported by Paul Thomas.
 
 1998-04-30  Ulrich Drepper  <drepper@cygnus.com>
 
diff -Nurp gettext-0.10.35.kn.2/m4/gettext.m4 gettext-0.10.35.kn.3/m4/gettext.m4
--- gettext-0.10.35.kn.2/m4/gettext.m4	Wed Sep 23 16:56:26 1998
+++ gettext-0.10.35.kn.3/m4/gettext.m4	Wed Sep 23 18:18:45 1998
@@ -218,8 +218,8 @@ AC_DEFUN(AM_GNU_GETTEXT,
 
    AC_CHECK_HEADERS([argz.h limits.h locale.h nl_types.h malloc.h string.h \
 unistd.h sys/param.h])
-   AC_CHECK_FUNCS([getcwd munmap putenv setenv setlocale strchr strcasecmp \
-strdup __argz_count __argz_stringify __argz_next])
+   AC_CHECK_FUNCS([getcwd memcpy munmap putenv setenv setlocale strchr \
+strcasecmp strdup __argz_count __argz_stringify __argz_next])
 
    if test "${ac_cv_func_stpcpy+set}" != "set"; then
      AC_CHECK_FUNCS(stpcpy)
